test/xpu_api/numerics/numeric.ops/inner.product/xpu_inner_product_comp.pass.cpp

```diff
@@ -14,144 +14,109 @@
+                                                    oneapi::dpl::multiplies<int>(), oneapi::dpl::plus<int>());
+                out[2] = oneapi::dpl::inner_product(Iter1(&in1[0]), Iter1(&in1[0] + 1), Iter2(&in2[0]), 1,
+                                                    oneapi::dpl::multiplies<int>(), oneapi::dpl::plus<int>());
+                out[3] = oneapi::dpl::inner_product(Iter1(&in1[0]), Iter1(&in1[0] + 1), Iter2(&in2[0]), 10,
+                                                    oneapi::dpl::multiplies<int>(), oneapi::dpl::plus<int>());
+                out[4] = oneapi::dpl::inner_product(Iter1(&in1[0]), Iter1(&in1[0] + 2), Iter2(&in2[0]), 1,
+                                                    oneapi::dpl::multiplies<int>(), oneapi::dpl::plus<int>());
+                out[5] = oneapi::dpl::inner_product(Iter1(&in1[0]), Iter1(&in1[0] + 2), Iter2(&in2[0]), 10,
+                                                    oneapi::dpl::multiplies<int>(), oneapi::dpl::plus<int>());
+                out[6] = oneapi::dpl::inner_product(Iter1(&in1[0]), Iter1(&in1[0] + 6), Iter2(&in2[0]), 1,
+                                                    oneapi::dpl::multiplies<int>(), oneapi::dpl::plus<int>());
+                out[7] = oneapi::dpl::inner_product(Iter1(&in1[0]), Iter1(&in1[0] + 6), Iter2(&in2[0]), 10,
+                                                    oneapi::dpl::multiplies<int>(), oneapi::dpl::plus<int>());
+            });
+        });
+    }
+    const int ref[8] = {1, 10, 7, 70, 49, 490, 117649, 1176490};
+    for (int i = 0; i < 8; ++i)
+    {
+        ASSERT_EQUAL(ref[i], output[i]);
+    }
 }
 
-class KernelTest1;
-class KernelTest2;
-class KernelTest3;
-class KernelTest4;
-class KernelTest5;
-class KernelTest6;
-class KernelTest7;
-class KernelTest8;
-class KernelTest9;
-class KernelTest10;
-class KernelTest11;
-class KernelTest12;
-class KernelTest13;
-class KernelTest14;
-class KernelTest15;
-class KernelTest16;
-class KernelTest17;
-class KernelTest18;
-class KernelTest19;
-class KernelTest20;
-class KernelTest21;
-class KernelTest22;
-class KernelTest23;
-class KernelTest24;
-class KernelTest25;
+int
+main()
+{
+    test<input_iterator<const int*>, input_iterator<const int*>>();
+    test<input_iterator<const int*>, forward_iterator<const int*>>();
+    test<input_iterator<const int*>, bidirectional_iterator<const int*>>();
+    test<input_iterator<const int*>, random_access_iterator<const int*>>();
+    test<input_iterator<const int*>, const int*>();
 
-int main() {
-  test<input_iterator<const int *>, input_iterator<const int *>, KernelTest1>();
-  test<input_iterator<const int *>, forward_iterator<const int *>,
-       KernelTest2>();
-  test<input_iterator<const int *>, bidirectional_iterator<const int *>,
-       KernelTest3>();
-  test<input_iterator<const int *>, random_access_iterator<const int *>,
-       KernelTest4>();
-  test<input_iterator<const int *>, const int *, KernelTest5>();
+#if !PSTL_USE_DEBUG
```

Comment:
Let's leave a comment that says that we turned out these test cases in debug mode since they consume too much time: here and in the other test.